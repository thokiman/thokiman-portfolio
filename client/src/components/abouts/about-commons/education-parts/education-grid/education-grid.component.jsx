import React from 'react';
import {
  EducationImage,
  EducationImageContainer,
  EducationImageItems,
} from './education-grid.styles';

export const EducationGrid = ({ imagePath }) => {
  return (
    <EducationImageContainer>
      {imagePath.map(({ id, field, iconPath }) => {
        return (
          <EducationImageItems className='education-image-items' key={id}>
            <EducationImage src={`${iconPath}`} alt={`${field}`} />
          </EducationImageItems>
        );
      })}
    </EducationImageContainer>
  );
};

export default EducationGrid;
