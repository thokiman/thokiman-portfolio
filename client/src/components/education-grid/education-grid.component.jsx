import React from "react";

import {
  EducationImage,
  EducationImageContainer,
  EducationImageItems,
} from "./education-grid.styles";

const EducationGrid = ({ imagePath }) => {
  return (
    <EducationImageContainer>
      {imagePath.map(({ id, field, iconPath }) => {
        return (
          <EducationImageItems key={id}>
            <EducationImage src={`${iconPath}`} alt={`${field}`} />
          </EducationImageItems>
        );
      })}
    </EducationImageContainer>
  );
};

export default EducationGrid;
