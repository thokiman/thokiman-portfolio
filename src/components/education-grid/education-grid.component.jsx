import React from "react";

import "./education-grid.styles.scss";

const EducationGrid = ({ imagePath }) => {
  return (
    <div className="education-image-container">
      {imagePath.map(({ id, field, iconPath }) => {
        return (
          <div key={id} className="education-image-items">
            <img
              className="education-image"
              src={`${iconPath}`}
              alt={`${field}`}
            />
          </div>
        );
      })}
    </div>
  );
};

export default EducationGrid;
