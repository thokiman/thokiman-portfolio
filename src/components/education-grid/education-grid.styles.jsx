import styled from "styled-components";

export const EducationImageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const EducationImageItems = styled.div`
  margin: 10px 0;
  width: 100%;
  height: 100%;
`;

export const EducationImage = styled.img`
  width: 100%;
  height: 70%;
  background-size: cover;
`;
EducationImageContainer.displayName = "EducationImageContainer";
EducationImageItems.displayName = "EducationImageItems";
EducationImage.displayName = "EducationImage";
