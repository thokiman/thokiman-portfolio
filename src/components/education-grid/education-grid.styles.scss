.education-image-container {

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    .education-image-items {

        margin: 10px 0;
        width: 100%;
        height: 100%;


        .education-image {

            width: 100%;
            height: 70%;
            background-size: cover;
        }
    }
}