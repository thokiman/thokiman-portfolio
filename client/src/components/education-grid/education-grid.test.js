import React from "react";
import { shallow } from "enzyme";

import { EducationGrid } from "./education-grid.component";

describe("EducationGrid component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      imagePath: mockArray,
    };
    wrapper = shallow(<EducationGrid {...mockProps} />);
  });

  it("should render EducationGrid component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
