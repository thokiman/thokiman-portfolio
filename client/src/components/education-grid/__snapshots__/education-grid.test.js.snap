// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationGrid component should render EducationGrid component 1`] = `ShallowWrapper {}`;
